### DISCUSSION TITLE
What is the function of the "log out" button in the startup menu of VNM?

### DISCUSSION THREAD

**Discussion by civier** [neurodesk/neurodesk.github.io-hugo#14] [Q&A] [answered]
I'm not sure about the function of the "log out" button in the startup menu of VNM. From my experience, after pressing "log out", it is impossible to connect to the desktop again until the docker container is stopped and started again.
If I check the status of the container after "log out" was pressed (with "docker ps"), it says that the container is an "unhealthy" state.
Do you have the same experience?

Ideally, it will be great if the "log out" button could be used to restart the container (if the user has a problem, things might fix after restarting, like in a physical computer), or if not possible, at least stop the container. For VNM running on a personal computer, it doesn't really matter, as VNM can be restarted using the docker CLI. However, for VNM running on the cloud, this can be very useful, as often users do not have access to docker CLI (in the setup we have in Swinburne, users do not have admin privileges for the cloud instance, and I start the VNM for them).

Whatever we do, it'll also be nice to describe the function of the "log out" button in the documentation.

**Comment by stebo85** [comment, marked as answer]
I think catching the unhealthy status and restarting the desktop could be possible. You could start in this repo: https://github.com/NeuroDesk/docker-ubuntu-vnc-desktop and see if you can detect this event in here: https://github.com/NeuroDesk/docker-ubuntu-vnc-desktop/blob/develop/rootfs/usr/local/lib/web/backend/vnc/state.py - submit a pull request with a proof-of-concept and we can merge it :)
